core/require.js

```diff
@@ -13,12 +13,21 @@
 'use strict';
 
 var hook = require('./unsafe/hook.js'),
-	preventCache = [];
+    preventCache = [],
+    inst = require('./install.js');
 
-global.requireHook = function(req) {
-    req.searchCache = function (moduleName, callback) {
-        var mod = req.resolve(moduleName);
-        if (mod && ((mod = req.cache[mod]) !== undefined)) {
+global.requireHook = function (req) {
+    var func = function (mod) {
+        return inst.requireOrInstall(req, mod);
+    };
+    for (var key in req) {
+        func[key] = req[key];
+    }
+    func.safe = func;
+
+    func.searchCache = function (moduleName, callback) {
+        var mod = func.resolve(moduleName);
+        if (mod && ((mod = func.cache[mod]) !== undefined)) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Unexpected use of undefined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530333308/issues/source?bid=3282987&fileBranchId=3282997#l30)